examples/apps/table-detail-tea/src/main.rs

```diff
@@ -0,0 +1,211 @@
+    panic::set_hook(Box::new(move |panic_info| {
+        stdout().execute(DisableMouseCapture).ok();
+        original_hook(panic_info);
+    }));
+}
+
+impl Default for AppModel {
+    fn default() -> Self {
+        Self {
+            people: (0..50).map(|_| Person::fake()).collect(),
+            table_state: TableState::default().with_selected(0),
+        }
+    }
+}
+
+impl AppModel {
+    fn run(mut self, mut terminal: DefaultTerminal) -> color_eyre::Result<()> {
+        loop {
+            terminal.draw(|frame| self.view(frame))?;
+
+            let mut maybe_message = handle_event()?;
+            while let Some(message) = maybe_message {
+                if message == Message::Quit {
+                    return Ok(());
+                }
+                maybe_message = self.update(message);
+            }
+        }
+    }
+
+    fn update(&mut self, message: Message) -> Option<Message> {
+        match message {
+            Message::Quit => {}
+            Message::SelectNext => self.table_state.select_next(),
+            Message::SelectPrevious => self.table_state.select_previous(),
+        };
+        None
+    }
+
+    /// Render the current view
+    fn view(&mut self, frame: &mut Frame) {
+        let [top, bottom] = Layout::vertical([Constraint::Fill(1); 2]).areas(frame.area());
+        self.render_table(frame, top);
+        self.render_detail(frame, bottom);
+    }
+
+    fn render_table(&mut self, frame: &mut Frame, area: Rect) {
+        let header = Row::new(vec!["Name", "Address", "Email"])
+            .style(Modifier::BOLD)
+            .height(1);
+        let rows = self.people.iter().map(Row::from);
+        let widths = [
+            Constraint::Length(20),
+            Constraint::Min(10),
+            Constraint::Fill(1),
+        ];
+        let table = Table::new(rows, widths)
+            .header(header)
+            .row_highlight_style(Style::new().add_modifier(Modifier::REVERSED | Modifier::ITALIC))
+            .block(Block::bordered().title(" People "));
+
+        frame.render_stateful_widget(table, area, &mut self.table_state);
+    }
+
+    fn render_detail(&self, frame: &mut Frame, area: Rect) {
+        let selected_item = &self.people[self.table_state.selected().unwrap()];
+        let block = Block::bordered()
+            .title(format!(" {} ", selected_item.name))
+            .title_bottom(" (Esc/q) quit | (↑) move up | (↓) move down ")
+            .padding(Padding::new(1, 1, 1, 1));
+        let text = format!("{}\n\n{}", selected_item.email, selected_item.address);
+        let detail = Paragraph::new(text).block(block);
+        frame.render_widget(detail, area);
+    }
+}
+
+/// Handle events and map to a Message
+fn handle_event() -> color_eyre::Result<Option<Message>> {
+    if !event::poll(Duration::from_millis(250))? {
+        return Ok(None);
+    }
+    match event::read()? {
+        Event::Key(key) if key.kind == event::KeyEventKind::Press => Ok(handle_key(key)),
```

Comment:
crossterm 0.29 change

```suggestion
        Event::Key(key) if key.is_press() => Ok(handle_key(key)),
```